### PR TITLE
188 sidebar functionality into utils

### LINKED ISSUE #188: Move functionality of modeler sidebar to utilities class
As an open source contributer, I'd like to reuse the functionality that has been created for the modeler sidebar in possible other interfaces. Therefore I'd prefer to have most of the functionality available in a utility class in order to use it.

---

## :white_check_mark: &nbsp; Conditions of satisfaction

- [ ] Should move functionality to at least one class in the utils folder
- [ ] Should functionality should be documented
- [ ] functionality of the sidebar should be preserved and use the newly created methods

### PR DESCRIPTION
**Pull request by kej-jay**
closes #188 

## Changes
The functionality of the sidebar is now available in separate methods from a utilities file. Additionally this functionality is now tested.

## Checklist before merge

**Developer's responsibilities**
* [ ] **Assign** one or two developers
* [ ] **Change code** if reviewer(s) has/have requested it
* [ ] **Pull request build** has passed
* [ ] **tested locally** (in at least chrome & firefox if frontend)
* [ ] updated the **documentation**
* [ ] added **tests** where necessary

### CODE REVIEW

File: client/src/components/content/ModelerSidebar/ModelerSidebar.jsx [hunk 1]
```diff
@@ -1,24 +1,21 @@
...
+  nameChangedHandler,
+  applicationChangedHandler,
+  taskChangedHandler,
+  handleInputParameterChange,
+  handleOutputVarNameChange,
+  downloadRobotFile,
+  onSaveToCloud,
+  onModelerSelectionChange,
+  onModelerElementChanged,
+} from '../../../utils/sidebarFunctionality';
```
Comment (lukashueller) [inline client/src/components/content/ModelerSidebar/ModelerSidebar.jsx:18]: Let's please define a consistent naming scheme for the methods. Maybe we talk about handlers for all methods or how do we get that named consistently?

File: client/src/utils/sidebarFunctionality.jsx [hunk 2]
```diff
@@ -0,0 +1,281 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable react/jsx-no-bind */
```
Comment (lukashueller) [inline client/src/utils/sidebarFunctionality.jsx:2]: we don't need this ESLint disabling anymore

**Review by lukashueller — COMMENTED** [review]
Please adress my comments. Otherwise it looks like a very clean job! Thanks!

File: client/src/utils/sidebarFunctionality.jsx [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+};
+
+/**
+ * @description Gets called when the the button is pressed to save to the cloud.
+ * This function will retrieve the xml from the parser, parse that xml to a ssot and write the
+ * resulting ssot into the sessionStorage.
+ * @param {Object} modeler the modeling object
+ * @param {String} robotId id of the robot
+ */
+const onSaveToCloud = async (modeler, robotId) => {
```
Comment (WolfgangDaniel) [inline client/src/utils/sidebarFunctionality.jsx:241]: This function is not exclusively used by the modeler sidebar and should therefore be put in an other utils file

**Review by WolfgangDaniel — CHANGES_REQUESTED** [review]
I really like the clean code and the high test coverage. No requests there. One general thing I would like to be changed though. I would like the file sidebarFunctionality.jsx to be called modelerSidebarFunctionality.js so that the name is directly linked to the component. 
Moreover I think it would be good to either have a components folder in utils and to put the functionality file in there or to put the functionality file right next to the component itself (so in this case next to the ModelerSidebar.jsx file)

File: client/src/utils/sidebarFunctionality.jsx (hunk 3 above)
Comment (kej-jay) [reply client/src/utils/sidebarFunctionality.jsx]: Download robot file would be another one of those methods potentially used also somewhere else. I pulled those two methods into a new file

**Review by WolfgangDaniel — APPROVED** [review]
Looks good!

**Review by lukashueller — APPROVED** [review]
Thanks for adressing my points! Love the new naming of the methods.
Please be careful while merging DEV into your branch before merging to DEV.

**Pull request merged**